### PR TITLE
Improve changelog management

### PR DESCRIPTION
**Pull request by Ndpnt**
- Improve changelog validation:
  - Ensure that it can be parsed
  - Ensure that funder credit is present
  - Ensure that changes are present and well formed in the `Unreleased` section
- Remove confusing `Unreleased` section title after releasing a version
- Auto add link to related pull request of a released version
- Allow to skip changelog validation by adding following section in the changelog:
```
## Unreleased [no-release]

 _Modifications made in this changeset do not alter its behavior or functionality from the perspective of the end user or other interacting systems. It does not add new features, does not enhance existing features, or alters the behavior of the software in a way that affects its use._
 ```

Remove dependency to external GitHub Action [superfaceai/release-changelog-action](https://github.com/superfaceai/release-changelog-action)

Fixes https://github.com/OpenTermsArchive/engine/issues/1009

### CODE REVIEW

**Comment by MattiSG** [comment]
As discussed synchronously with @ndpnt and @clementbiron, the current implementation has the following limitations:

1. Contributors will receive a test failure until a maintainer adds the `No Release` label.
2. Maintainers will need to manually trigger a test run after adding the `No Release` label.
3. Increased vendor lock-in with GitHub.

An alternative could be to offer a line to copy-paste in the `Unreleased` section, in which the contributor certifies that, to the best of their knowledge, the contribution has no functional impact and should not trigger a release. Such an implementation would have the following limitations:

1. Provide a green check mark for the `check changelog` check when the line is present, leaving more room for an accidental lack of release from a tired maintainer.
2. Add a commit erasing the “Unreleased” section after each non-functional branch merge.
3. Force an unusual action from contributors: copy-pasting a specific line from the CONTRIBUTING file.

File: CONTRIBUTING.md
Comment (MattiSG) [inline CONTRIBUTING.md:70]: This is great, but the two sentences are a bit redundant 🙂 
```suggestion
_Modifications made in this changeset do not add, remove or alter any behavior, dependency, API or functionality of the software. They only change non-functional parts of the repository, such as the README file or CI workflows._
```

File: CONTRIBUTING.md
Comment (MattiSG) [inline CONTRIBUTING.md:68]:
```suggestion
## Unreleased [no-release]
```

File: package.json
Comment (MattiSG) [inline package.json:85]: Why is this in `dependencies` and not `devDependencies`?

**Review by MattiSG — COMMENTED** [review]
(partial review)

File: scripts/changelog/index.js
Comment (MattiSG) [inline scripts/changelog/index.js:15]:
```suggestion
  .description('A command-line utility for managing the changelog file')
```

File: CONTRIBUTING.md
Comment (MattiSG) [inline CONTRIBUTING.md:75]:
```suggestion
4. Since each release is produced automatically from a single pull request, the [notice](https://common-changelog.org/#23-notice) is always used to link to the source pull request rather than [references](https://common-changelog.org/#242-references), which would always reference the same pull request. References can instead link to relevant specific parts of an RFC, decision record or diff. This notice is automatically generated in CI by the release process.
```

File: CONTRIBUTING.md
Comment (MattiSG) [inline CONTRIBUTING.md:65]:
```suggestion
For non-functional changes (documentation, CI workflow…) that should not trigger a release, the following content should be added to the changelog:
```

File: .github/workflows/release.yml
Comment (MattiSG) [inline .github/workflows/release.yml:102]:
```suggestion
      - name: Erase unreleased information from changelog
```

File: scripts/changelog/changelog.test.js
Comment (MattiSG) [inline scripts/changelog/changelog.test.js:55]:
```suggestion
    context('when getting a non-existing version', () => {
```

**Review by MattiSG — CHANGES_REQUESTED** [review]

File: scripts/changelog/index.js
Comment (MattiSG) [inline scripts/changelog/index.js:45]: Shouldn't we check that a version number was provided (as opposed to the option being accidentally used as a Boolean flag)? But maybe this is not needed anymore if the docopt syntax is corrected as commander will validate the existence of a parameter 🙂
```suggestion
if (options.getVersionContent instance of String) {
```

**Review by MattiSG — CHANGES_REQUESTED** [review]
Almost there, but the checks and the option syntax should be corrected.

File: .github/workflows/changelog.yml
Comment (Ndpnt) [reply .github/workflows/changelog.yml:51]: Yep, but when release workflows fails it is really useful to have this information, so I suggest to let it there for futur debugging

File: scripts/changelog/changelog.js
Comment (Ndpnt) [reply scripts/changelog/changelog.js:7]: - Multiline is required to have `^` and `$` to match the begin/end of each line (not only begin/end of string). This is essential as we analyze the entire release content to verify the funder.
- We have an issue with the specification regarding certain release funders formatted as follows: `> Development of this release was made on a volunteer basis by ….` Should we modify these types of funder sentences, or should we adjust the specification accordingly?

Comment (MattiSG) [reply scripts/changelog/changelog.js:7]: We should adjust the spec as volunteer contributions are an integral part of the contribution workflow.

Gotcha for the multiline, I had misunderstood this flag as valid for the match, not for the input.

**Pull request merged**
